curriculum/challenges/english/22-rosetta-code/rosetta-code-challenges/tokenize-a-string-with-escaping.md

```diff
@@ -98,27 +119,42 @@ function tokenize(str, sep, esc) {
 ```js
 // tokenize :: String -> Character -> Character -> [String]
 function tokenize(str, charDelim, charEsc) {
-  const dctParse = str.split('')
-    .reduce((a, x) => {
-      const blnEsc = a.esc;
-      const blnBreak = !blnEsc && x === charDelim;
-      const blnEscChar = !blnEsc && x === charEsc;
-
-      return {
-        esc: blnEscChar,
-        token: blnBreak ? '' : (
-          a.token + (blnEscChar ? '' : x)
-        ),
-        list: a.list.concat(blnBreak ? a.token : [])
-      };
-    }, {
-      esc: false,
-      token: '',
-      list: []
-    });
-
-  return dctParse.list.concat(
-    dctParse.token
-  );
+  let parts = [];
+  let currentPart = '';
+  let isEscaped = false;
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i];
+    if (isEscaped) {
+      currentPart += char;
+      isEscaped = false;
+    } else if (char === charEsc) {
+      isEscaped = true;
+    } else if (char === charDelim) {
+      parts.push(currentPart);
+      currentPart = '';
+    } else {
+      currentPart += char;
+    }
+  }
+  parts.push(currentPart); 
+  return parts;
 }
+
+const testCases = [
+  { str: testStr1, sep: '|', esc: '^', expected: res1 },
+  { str: testStr2, sep: '&', esc: '@', expected: res2 },
+  { str: testStr4, sep: '|', esc: '^', expected: res4 },
+  { str: testStr5, sep: '|', esc: '^', expected: res5 },
+  { str: testStr6, sep: '|', esc: '^', expected: res6 },
+  { str: testStr7, sep: '|', esc: '^', expected: res7 },
+  { str: testStr8, sep: '|', esc: '^', expected: res8 },
+  { str: testStr9, sep: '|', esc: '^', expected: res9 }
+];
+testCases.forEach((testCase, index) => {
+  console.log(`Test Case ${index + 1}:`);
+  console.log(`Input String: '${testCase.str}'`);
+  console.log(`Expected Output:`, testCase.expected);
+  console.log(`Actual Output:`, tokenize(testCase.str, testCase.sep, testCase.esc));
+  console.log('----------------------------------------');
+});
```

Comment:
I don't see a need to add these to the solution. It's extra bloat for something that's only used in our CI pipeline.